test/integration/integration.test.ts

```diff
@@ -0,0 +1,605 @@
+import { beforeAll, describe, expect, it } from 'vitest';
+import { createClients } from './utilities/setup.js';
+import { LocationsEventMap } from '../../src/Locations.js';
+import { SpaceEventMap } from '../../src/Space.js';
+import { MembersEventMap } from '../../src/Members.js';
+import { CursorsEventMap } from '../../src/Cursors.js';
+import { nanoid } from 'nanoid';
+import { LocksEventMap } from '../../src/Locks.js';
+
+describe(
+  'integration tests',
+  () => {
+    describe('space members', () => {
+      let performerSpaces;
+      let observerSpaces;
+      let performerClientId;
+      let performerSpace;
+      let observerSpace;
```

Comment:
One final remark @lawrence-forooghian - do you think it's worth outlining the approach taken in this test at the top of the file, similar to what you have done in the commit? I think it might not be obvious that each `it` is treated as a step in the spec instead of an individual test, and they share state in a `describe` block.

Reply:
Sure, have done.